src/Network/Network/NatGateway/NewAzureRMNatGatewayCommand.cs

```diff
@@ -50,6 +50,11 @@ public partial class NewAzureRmNatGateway : NetworkBaseCmdlet
             HelpMessage = "The idle timeout of the nat gateway.")]
         public int IdleTimeoutInMinutes { get; set; } = 4;
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A list of availability zones denoting the zone in which Nat Gateway should be deployed.")]
+        public string[] Zone { get; set; }
```

Comment:
Are availability zones numbers (like ints) or are there zones that have alpha characters? It would make sense to be int if they are only numbers.

Reply:
[Per API specs](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/natGateway.json#L400) it is array of strings. Also it is consistent across Network PS: [sample](https://github.com/Azure/azure-powershell/blob/master/src/Network/Network/PublicIpAddress/NewAzurePublicIpAddressCommand.cs#L124)

Reply:
If availability zones are simply numbers, I'd recommend the cmdlet use numbers, since it correctly specifies the expected input. I will likely make this change for the new Network generated cmdlets too.

Reply:
Currently there are three zones represented as "1", "2", "3" (they are not Network-specific, [Compute](https://github.com/Azure/azure-powershell/blob/master/src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs#L107) uses them too) but there is always a chance more zones are introduced that could use alphabet characters.

Reply:
@number213 After some discussion, using `string` is fine as long as there is some example documentation showing what values make sense. In the test above, the usage of this parameter is `-Zone 1` and not `-Zone '1'` or `-Zone "1"`. That's what lead to this discussion to begin with. A completer would also be appropriate too, whichever you think is better.